lib/r10k/api.rb

```diff
@@ -88,72 +91,78 @@ def parse_puppetfile(io_or_content)
     # @raise [NotImplementedError] Control repo source has a type that is not currently supported.
-    # @raise [RuntimeError] Something bad happened, see exception message.
-    def envmap_from_source(source, env_name, base_cachedir)
+    # @raise [RuntimeError]
+    def envmap_from_source(control_source, env_name, opts={})
       # TODO: enforce valid Puppet environment name here? verify prefix? maybe just use R10K::Environment::Name?
       branch_name = env_name
 
-      if source[:prefix]
-        if source[:prefix].is_a? String
-          branch_name = env_name.gsub(/^#{Regexp.quote(source[:prefix])}_/, '')
+      if control_source[:prefix]
+        if control_source[:prefix].is_a? String
+          branch_name = env_name.gsub(/^#{Regexp.quote(control_source[:prefix])}_/, '')
         else
-          branch_name = env_name.gsub(/^#{Regexp.quote(source[:name])}_/, '')
+          branch_name = env_name.gsub(/^#{Regexp.quote(control_source[:name])}_/, '')
         end
       end
 
-      case source[:type].to_sym
+      case control_source[:type].to_sym
       when :git
-        cachedir = cachedir_for_git_remote(source[:remote], base_cachedir)
+        git_dir = cachedir_for_git_remote(control_source[:remote], opts[:cachedir])
 
         begin
-          version = git.rev_parse(branch_name, git_dir: cachedir)
+          commit_sha = git.resolve_commit(git_dir, branch_name)
         rescue R10K::Git::GitError => e
           raise RuntimeError.new("Unable to resolve branch name '#{branch_name}' to a Git commit: #{e.message}")
         end
 
         # TODO: figure out how to pass through base_path option
-        puppetfile = get_puppetfile(:git, cachedir, version)
+        puppetfile = get_puppetfile(source version)
       when :svn
         raise NotImplementedError
       else
         raise RuntimeError.new("Unrecognized control repo source type: #{source[:type]}")
       end
 
-      envmap = {
+      return {
         environment: env_name,
-        source: source,
+        source: control_source,
         version: version,
         resolved_at: nil,
+        modules: parse_puppetfile(puppetfile),
       }
-
-      envmap.merge(parse_puppetfile(puppetfile))
     end
 
     # Creates a resolved environment hashmap representing the actual state of the Puppet environment found at the given path.
     #
     # @param path [String] Path on disk of the Puppet environment to be inspected.
-    # @param moduledir [String] The path, relative to the environment path, where the modules are deployed.
+    # @options opts [String] :moduledir The path, relative to the environment, where modules are deployed. (Default: "modules")
     # @return [Hash] A hashmap representing the actual state of the environment found at path.
     # @raise RuntimeError
-    def parse_deployed_env(path, moduledir="modules")
+    def parse_deployed_env(path, opts={})
+      moduledir = opts[:moduledir] || "modules"
```

Comment:
"modules" here feels like it should be a constant, since it's used in a couple of places in the class.